composer.json

```diff
@@ -24,7 +24,12 @@
         "phpunit/phpunit": "^7.0",
         "doctrine/coding-standard": "^6.0",
         "phpstan/phpstan-shim": "^0.9.2",
-        "vimeo/psalm": "^3.2.2"
+        "vimeo/psalm": "^3.8.1"
+    },
+    "config": {
+        "platform": {
+            "php": "7.2.0"
```

Comment:
Why this @greg0ire ?

Reply:
Oh sorry I meant to use 7.1. The reason for this is being able to update `vimeo/psalm` without ending up with 7.4 dependencies, and a red build: https://travis-ci.org/github/doctrine/collections/builds/690553813

Reply:
I will use `7.1.3` here, since that's what Psalm requires and it still is ok with what we use in Travis: (7.1.11)

Reply:
To be more clear: my goal is that people needing to update our deps like in this PR can do so safely from machines that use php 9 or whatever.

Reply:
I thought config.platform is in effect only when it's root project, not as dependency. If you want to allow doctrine/collections to install on php 8, we should change php constraint

Reply:
> If you want to allow doctrine/collections to install on php 8, we should change php constraint

I don't want that. All I want is for a contributor that uses another version of php than 7.1 to be able to _generate_ a composer.lock that is guaranteed to work on 7.1, because that is the lowest php version we use on Travis. See failing build above for why this is necessary.

Reply:
Oh my god, again this lockfile bs and working around it. We should seriously drop it.